lib/Doctrine/ORM/Tools/Console/Command/ValidateSchemaCommand.php

```diff
@@ -76,7 +76,9 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         if ($input->getOption('skip-mapping')) {
             $output->writeln('<comment>[Mapping]  Skipped mapping check.</comment>');
-        } elseif ($errors = $validator->validateMapping()) {
+        } elseif (($errors = $validator->validateMapping()) === false) {
+            $output->writeln('<error>[Mapping]  FAIL - No entities defined!</error>');
+        } elseif (count($errors) > 0) {
```

Comment:
I suggest counting `$em->getMetadataFactory()->getAllMetadata()` and dealing with the case where 0 mappings are found directly in the CLI command. The validator context is much more specific, and an empty set of metadata is indeed valid.